### PR TITLE
refactor(wallet): Streamline lockdown process using consoleTaming param

### PR DESCRIPTION
**Pull request by samsiegart**
Previously lockdown was handled separately between dev and prod (context: https://github.com/Agoric/agoric-sdk/pull/3879).

This process can be simplified by passing `consoleTaming: 'unsafe'` to `lockdown` in dev mode. This allows the react framework to properly extend the `console` object for its debugging features.

### CODE REVIEW

File: packages/dapp-svelte-wallet/react-ui/public/index.html [hunk 1]
```diff
@@ -4,12 +4,22 @@
...
+      // features.
+      const consoleTaming = '%NODE_ENV%' === 'production' ? 'safe' : 'unsafe';
+      
+      const { pow: mathPow } = Math;
+      Math.pow = (base, exp) => 
+        (typeof base === 'bigint' && typeof exp ==='bigint') 
+        ? base ** exp : mathPow(base, exp);
+      
+      lockdown({ 
+        __allowUnsafeMonkeyPatching__: 'unsafe',
```
Comment (erights) [inline packages/dapp-svelte-wallet/react-ui/public/index.html:17]: Do we still need this? Please try removing and let me know what breaks.

File: packages/dapp-svelte-wallet/react-ui/public/index.html [hunk 2]
```diff
@@ -4,12 +4,22 @@
...
     <script>
-      if ('%NODE_ENV%' === 'production') {
-        const { pow: mathPow } = Math;
-        Math.pow = (base, exp) => (typeof base === 'bigint' && typeof exp ==='bigint') ? base ** exp : mathPow(base, exp);
-        lockdown({ __allowUnsafeMonkeyPatching__: 'unsafe', errorTaming: 'unsafe', overrideTaming: 'severe' });
-        console.log('lockdown done.');
-      }
+      // Allow the React dev environment to extend the console for debugging
+      // features.
+      const consoleTaming = '%NODE_ENV%' === 'production' ? 'safe' : 'unsafe';
```
Comment (erights) [inline packages/dapp-svelte-wallet/react-ui/public/index.html:9]: Not great, but ok for now.

**Review by erights — CHANGES_REQUESTED** [review]
The `consoleTaming` approach is adequate for now.

I'm "Request changes"ing because of the `__allowUnsafeMonkeyPatching__`. What happens if this is removed?

What's the easiest way for me to reproduce locally the problems that this PR fixes?

File: packages/dapp-svelte-wallet/react-ui/public/index.html (hunk 1 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/react-ui/public/index.html:17]: In prod, changing this  to 'safe' gives the following:
```
Uncaught TypeError: Cannot define property decorate, object is not extensible
  at Function.defineProperty (<anonymous>)
  at injectedScript.bundle.js:67
  at injectedScript.bundle.js:67
  at injectedScript.bundle.js:67
  at injectedScript.bundle.js:67
  at Object.<anonymous> (injectedScript.bundle.js:67)
  at Object.<anonymous> (injectedScript.bundle.js:67)
  at r (injectedScript.bundle.js:1)
  at Object.<anonymous> (injectedScript.bundle.js:25)
  at r (injectedScript.bundle.js:1)
  ```
In dev, the above occurs as well as:
```
Uncaught TypeError: Cannot assign to read only property 'Symbol(Symbol.iterator)' of object '[object Generator]'
  at index.js:1
  at Object.<anonymous> (index.js:1)
  at r (index.js:1)
  at Object.<anonymous> (index.js:1)
  at r (index.js:1)
  at Module.<anonymous> (index.js:1)
  at r (index.js:1)
  at index.js:1
  at index.js:1
  at index.js:1
  ```

**Comment by samsiegart** [comment]
> The `consoleTaming` approach is adequate for now.
> 
> I'm "Request changes"ing because of the `__allowUnsafeMonkeyPatching__`. What happens if this is removed?
> 
> What's the easiest way for me to reproduce locally the problems that this PR fixes?

The problem this fixes is, for **dev** only, having to call lockdown in index.js (which runs after react does all the finagling with console). With `consoleTaming: 'unsafe'`, we can run lockdown before any other scripts on the page load.

You can checkout this branch and run local dev mode with `cd agoric-sdk/packages/dapp-svelte-wallet`, `agoric install`, `cd react-ui`, `yarn install` and `yarn start`. Then you can experiment with modifying the `lockdown` arguments in `index.html`.

**Review by erights — APPROVED** [review]
Thanks for the answers. LGTM

**Pull request merged**